middlewares/authenticate.go

```diff
@@ -139,15 +139,20 @@ func (m *AuthenticateMiddleware) tryGetUserByApiKeyQuery(r *http.Request) (*mode
 }
 
 func (m *AuthenticateMiddleware) tryGetUserByTrustedHeader(r *http.Request) (*models.User, error) {
+	if !m.config.Security.TrustedHeaderAuth {
+		return nil, errors.New("trusted header auth disabled")
+	}
+
 	remoteUser := r.Header.Get(m.config.Security.TrustedHeaderAuthKey)
 	if remoteUser == "" {
 		return nil, errors.New("trusted header field empty")
 	}
-	if addr, err := net.ResolveTCPAddr("tcp", r.RemoteAddr); err != nil || !slice.ContainBy[net.IP](m.config.Security.TrustReverseProxyIPs(), func(ip net.IP) bool {
-		return addr.IP.Equal(ip)
+	if addr, err := net.ResolveTCPAddr("tcp", r.RemoteAddr); err != nil || !slice.ContainBy[net.IPNet](m.config.Security.TrustReverseProxyIPs(), func(ipNet net.IPNet) bool {
```

Comment:
> So, if I understand correctly, when `m.config.Security.TrustedHeaderAuth` is set, the TCP remote IP is checked against the trust list. If the trust list is not defined, all requests get blocked.

Yes, if `m.config.Security.TrustedHeaderAuth` is set to `true`, but the trusted IP list itself is empty, all request will be blocked.

> I would have expected that all requests containing X-Forwarded-For or X-Real-IP are checked against a trust list by default. That would be an even more secure default.

Fair point. Explicitly whitelisting reverse proxies by their IPs would be another security measure (but also a breaking change if not implemented as opt-in) we could consider implementing at some point. It would serve a slightly different purpose, though. The [trusted header auth.](https://github.com/muety/wakapi/issues/534) feature covered in this PR is all about _authenticating_ users, i.e. "automatically" logging a user in to Wakapi based on what identity the reverse proxy assign to them (`Remote-User` header), while what you proposed is about distrusting _any_ sort of proxied request by default.